Rachaelbond patch 2

rachaelbond opened the pull request:
Makes the product image size selectable in config_flow. 0 to disable and either 100, 200, or 400 px width

Anrolosia commented:
It's a good idea to do that, thanks for the PR
One thing tho, instead of setting the default value for everyone to 0, could you use the migrate entry function to set the old size (100) if the checkbox was checked? And of course you'll have to increase the config version

rachaelbond commented:
Hi @Anrolosia I *think* this deals with the new config change in __init__.py correctly, but please double-check! I've not done a home-assistant config migration before. I've also changed the config variable name from "adding_images" -> "image_download_size" and removed "adding_images" in the migration. The other files have been updated to reflect the new config variable name.

rachaelbond commented:
Silly mistake corrected. The default value of "image_download_size" when imported should be 0, not TRUE

Anrolosia commented on custom_components/shopping_list_with_grocy/config_flow.py:
```diff
@@ -130,7 +130,8 @@ async def async_step_user(self, user_input=None):
                     vol.Required("mqtt_password"): cv.string,
-                    vol.Optional("adding_images", default=True): cv.boolean,
+                    vol.Optional("image_download_size", default="0"):
+                        vol.All(cv.string, vol.In(["0", "100", "200", "400"])),
```
I think it could be better to use integers instead of strings

Anrolosia reviewed (commented):
Globaly ok, minus my comment on integers instead of strings. Could you please take a look and adapt other files?

rachaelbond reviewed (commented):
Changed to using an integer

rachaelbond reviewed (commented):
As above

rachaelbond reviewed (commented):
As above

rachaelbond reviewed (commented):
Update  es.json for #13
